gnssanalysis/gn_download.py

```diff
@@ -846,325 +848,169 @@ def download_multiple_files_from_cddis(files: [str], ftp_folder: str, output_fol
         list(executor.map(download_file_from_cddis, files, _repeat(ftp_folder), _repeat(output_folder)))
 
 
-# TODO: Deprecate? Only supports legacy filenames
-def download_prod(
-    dates,
-    dest,
-    ac="igs",
-    suff="",
-    f_type="sp3",
-    dwn_src="cddis",
-    ftps=False,
-    f_dict=False,
-    wkly_file=False,
-    repro3=False,
-):
+def download_product_from_cddis(
+    download_dir: _Path,
+    start_epoch: _datetime,
+    end_epoch: _datetime,
+    file_ext: str,
+    limit: int = None,
+    long_filename: bool = False,
+    analysis_center: str = "IGS",
+    solution_type: str = "ULT",
+    sampling_rate: str = "15M",
+    project_type: str = "OPS",
+    timespan: _datetime.timedelta = _datetime.timedelta(days=2),
+    if_file_present: str = "prompt_user",
+) -> None:
     """
-    Function used to get the product file/s from download server of choice, default: CDDIS
-
-    Input:
-    dest - destination (str)
-    ac - Analysis Center / product of choice (e.g. igs, igr, cod, jpl, gfz, default = igs)
-    suff - optional suffix added to file name (e.g. _0 or _06 for ultra-rapid products)
-    f_type - file type to download (e.g. clk, cls, erp, sp3, sum, default = sp3)
-    dwn_src - Download Source (e.g. cddis, ga)
-    ftps - Optionally input active ftps connection object
-    wkly_file - optionally grab the weekly file rather than the daily
-    repro3 - option to download the REPRO3 version of the file
-
+    Download the file/s from CDDIS based on start and end epoch, to the
+    provided the download directory (download_dir)
     """
+    # DZ: Download the correct IGS FIN ERP files
+    if file_ext == "ERP" and analysis_center == "IGS" and solution_type == "FIN":  # get the correct start_epoch
+        start_epoch = GPSDate(str(start_epoch))
+        start_epoch = gpswkD2dt(f"{start_epoch.gpswk}0")
+        timespan = _datetime.timedelta(days=7)
+    # Details for debugging purposes:
+    logging.debug("Attempting CDDIS Product download/s")
+    logging.debug(f"Start Epoch - {start_epoch}")
+    logging.debug(f"End Epoch - {end_epoch}")
+
+    reference_start = _deepcopy(start_epoch)
+    product_filename, gps_date, reference_start = generate_product_filename(
+        reference_start,
+        file_ext,
+        long_filename=long_filename,
+        AC=analysis_center,
```

Comment:
Not sure what happened here
```suggestion
        analysis_center=analysis_center,
```